### PR TITLE
AMB-1290: C# User Restricted Separate Auth (CIS2)

### PR DESCRIPTION
**Pull request by ChewieCB**

### CODE REVIEW

File: user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Pages/Login.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Authentication;
+
+// Specifies access to this page requires authentication. Triggers OIDC flow if user not authenticated
+[Authorize]
+public class LoginModel : PageModel
+{
+    private readonly IConfiguration _configuration;
+    private readonly ILogger<IndexModel> _logger;
+    private readonly JwtHandler _jwtHandler;
```
Comment (artronics) [inline user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Pages/Login.cshtml.cs:16]: This is unused.

File: user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Pages/Error.cshtml.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+{
+    [ResponseCache(Duration = 0, Location = ResponseCacheLocation.None, NoStore = true)]
+    [IgnoreAntiforgeryToken]
+    public class ErrorModel : PageModel
+    {
+        public string? RequestId { get; set; }
+
+        public bool ShowRequestId => !string.IsNullOrEmpty(RequestId);
+
+        private readonly ILogger<ErrorModel> _logger;
```
Comment (artronics) [inline user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Pages/Error.cshtml.cs:15]: I noticed pretty all `_logger` fields are unused. I'm not sure if it's framework magic and requires that or they can be safely deleted.

**Comment by artronics** [comment]
Jack I ran code analysis with Rider. Here is a screenshot for your reference. I don't think all of  them need addressing. I'll comment ones that are important
image

File: user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Program.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+namespace HelloWorldSeparateAuth;
```
Comment (artronics) [inline user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Program.cs:1]: This entire file is unused.

File: user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Startup.cs [hunk 4]
```diff
@@ -0,0 +1,64 @@
+using HelloWorldSeparateAuth.JWT;
```
Comment (artronics) [inline user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Startup.cs:1]: This is unused.

File: user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Program.cs (hunk 3 above)
Comment (ChewieCB) [reply user-restricted-separate-auth-tutorials/cis2/csharp/HelloWorldSeparateAuth/Program.cs:1]: Program.cs is the default entry point for dotnet applications, it's also used to make the access token accessible to multiple namespaces via the `Config` class

**Review by RyanThomas1214 — APPROVED** [review]

**Pull request merged**